src/Conversations/Conversation.php

```diff
@@ -59,20 +59,29 @@ public function messages()
      *
      * @return Message
      */
-    public function getMessages($user, $perPage = 25, $page = 1, $sorting = 'asc', $columns = ['*'], $pageName = 'page')
-    {
-        return $this->messages()
-            ->join('notifications', 'notifications.data->message_id', '=', 'mc_messages.id')
-            ->where('notifications.notifiable_id', $user->id)
-            ->orderBy('mc_messages.id', $sorting)
-            ->paginate(
-                $perPage,
-                ['notifications.read_at', 'notifications.notifiable_id', 'notifications.id as notification_id',
-                    'mc_messages.*', ],
-                $pageName,
-                $page
-            );
-    }
+     public function getMessages($user, $perPage = 25, $page = 1, $sorting = 'asc', $columns = ['*'], $pageName = 'page')
+     {
+         $connection = config('database.default');
+         $driver = config("database.connections.{$connection}.driver");
+ 
+         $query  = $this->messages()
+             ->join('notifications', function($join) use ($driver) {
+                 $join->where('notifications.type', 'Musonza\Chat\Notifications\MessageSent')
+                      ->on($driver == 'sqlsrv' ? DB::raw('JSON_VALUE(data, \'$.message_id\')') :
+                          'notifications.data->message_id', '=', 'mc_messages.id');
+             })
+             ->where('notifications.notifiable_id', $user->user_id)
```

Comment:
Everything else looks fine but `->where('notifications.notifiable_id', $user->user_id)` still needs to have `$user->id` instead of `$user->user_id` to get the id of the user without any knowledge of the conversations they are in. Does this make sense?

Reply:
You're right. i'm updating my changes. i'll add in the config file something like: "User primary key" to let users write "id" or "user_id". And i'm making the $driver var a class prop. I will push my changes soon. Thank you.